ghcide/src/Text/Fuzzy/Parallel.hs

```diff
@@ -102,6 +102,40 @@ simpleFilter :: Int      -- ^ Chunk size. 1000 works well.
 simpleFilter chunk maxRes pattern xs =
   filter chunk maxRes pattern xs id
 
+
+-- | The function to filter a list of values by fuzzy search on the text extracted from them.
+filter' :: Int           -- ^ Chunk size. 1000 works well.
+       -> Int           -- ^ Max. number of results wanted
+       -> T.Text        -- ^ Pattern.
+       -> [t]           -- ^ The list of values containing the text to search in.
+       -> (t -> T.Text) -- ^ The function to extract the text from the container.
+       -> (T.Text -> T.Text -> Maybe Int) -- ^ Function to use for matching
```

Comment:
Well, when filtering which completions I want, I couldn't figure out a way to match based on anything but how close the words are to the already partially written word. 
But in the case of licenses I want to be able to tell the filter to consider the popular licenses more strongly, for example `MIT` should be suggested when `M` has been written but it is drowned out by `MIT-0` etc. Therefore, I wanted to be able to add a weight for each potential completion which supports the more popular licenses and guarantees they are in the list of suggestions. 
The function, I added as a parameter is a scoring function which can be used as an alternative calculation to the `closeness` score between the partially written word and the potential completion. It is duplicated code since I thought changing the original `simpleFilter` would be too intrusive and changing the `filter`called by the original `simpleFilter` might impact performance unexpectedly.

Reply:
Okay, but we want to write this down :) 

I think rewriting the original function to use the new function should be fine. If you're worried, you could mark the usage in `filter` with `GHC.Magic.Inline`. Then GHC should eliminate the overhead from the `Maybe`, I think. You could check by looking at the Core.